File: sBuilding/Trunk/Client/sBuilding/sTimeKeeping/Interface/ITimeKeepingDeviceConfig.cs
using sBuildingCommunication.Model;
using sTimeKeeping.Model;
using sWorldModel.TransportData;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace sTimeKeeping.Interface
{
    /// <summary>
    /// interface ITimeKeepingDeviceConfig
    /// </summary>
    public interface ITimeKeepingDeviceConfig
    {
        /// <summary>
        /// Insert DeviceConfig
        /// </summary>
        /// <param name="session"></param>
        /// <param name="deviceConfig"></param>
        /// <returns></returns>
        DeviceConfig InsertDeviceConfig(string session, DeviceConfig deviceConfig);
        /// <summary>
        /// Update DeviceConfig
        /// </summary>
        /// <param name="session"></param>
        /// <param name="deviceConfig"></param>
        /// <returns></returns>
        DeviceConfig UpdateDeviceConfig(string session, DeviceConfig deviceConfig);
        /// <summary>
        /// Delete DeviceConfig
        /// </summary>
        /// <param name="session"></param>
        /// <param name="deviceConfigId"></param>
        /// <returns></returns>
        int DeleteDeviceConfig(string session, long deviceConfigId);
        /// <summary>
        /// Get Device Config By Id
        /// </summary>
        /// <param name="session"></param>
        /// <param name="deviceConfigId"></param>
        /// <returns></returns>
        DeviceConfig GetDeviceConfigById(string session, long deviceConfigId);
        /// <summary>
        /// Insert Device By OrgId
        /// </summary>
        /// <param name="session"></param>
        /// <param name="orgId"></param>
        /// <param name="lstDevice"></param>
        /// <returns></returns>
        int InsertDeviceByOrgId(string session, long orgId, List<DeviceConfig> lstDevice);
        /// <summary>
        /// Delete List DeviceConfig
        /// </summary>
        /// <param name="session"></param>
        /// <param name="lstIdDevice"></param>
        /// <returns></returns>
        int DeleteListDeviceConfig(string session, List<long> lstIdDevice);
        /// <summary>
        /// Get ListDeviceConfig By OrgId
        /// </summary>
        /// <param name="session"></param>
        /// <param name="orgId"></param>
        /// <returns></returns>
        List<DeviceDoor> GetListDeviceConfigByOrgId(string session, long orgId);
        /// <summary>
        /// Check Ip Device For TimeKeeping
        /// </summary>
        /// <param name="session"></param>
        /// <param name="serial"></param>
        /// <param name="deviceconfigip"></param>
        /// <returns></returns>
        TimeKeepingAcessDTO CheckIpDeviceForTimeKeeping(string session, string serial, string deviceconfigip);
    }
}

File: sBuilding/Trunk/Server/Java/timekeepingservice/src/com/swt/timekeeping/ITimeKeepingDeviceConfig.java
package com.swt.timekeeping;

import java.util.List;

import com.swt.timekeeping.domain.DeviceConfig;

/**
 * ITimeKeepingDeviceConfig interface
 * 
 * @author TrangPig
 *
 */
public interface ITimeKeepingDeviceConfig {
	/**
	 * insert TimeKeepingDeviceConfig
	 * 
	 * @param timeKeepingConfig
	 * @return
	 */
	public DeviceConfig insert(DeviceConfig timeKeepingConfig);

	/**
	 * update TimeKeepingDeviceConfig
	 * 
	 * @param timeKeepingConfig
	 * @return
	 */
	public DeviceConfig update(DeviceConfig timeKeepingConfig);

	/**
	 * delete TimeKeepingDeviceConfig
	 * 
	 * @param timeKeepingConfigId
	 * @return
	 */
	public int delete(long timeKeepingConfigId);
	
	DeviceConfig getDeviceConfigById(long idDeviceConfig);

	/**
	 * getTimeKeepingConfigById
	 * 
	 * @param timeKeepingConfigId
	 * @return
	 */
	public DeviceConfig getTimeKeepingConfigById(long timeKeepingConfigId);
	
	public List<DeviceConfig> getTimeKeepingConfigByIp(String timeKeepingConfigIp);
	
	public List<DeviceConfig> getListDeviceConfigByOrgId(long orgId);

}
